addons/website_event_track/controllers/event.py

```diff
@@ -114,10 +86,7 @@ def event_tracks(self, event, tag=None, **post):
         searches = {}
         if tag:
             searches.update(tag=tag.id)
-            track_obj = request.registry.get('event.track')
-            track_ids = track_obj.search(request.cr, request.uid,
-                [("id", "in", [track.id for track in event.track_ids]), ("tag_ids", "=", tag.id)], context=request.context)
-            tracks = track_obj.browse(request.cr, request.uid, track_ids, context=request.context)
+            tracks = request.env['event.track'].search([("id", "in", event.track_ids.ids), ("tag_ids", "=", tag.id)])
```

Comment:
here use recordset.filtered method like

event.track_ids.filtered(lambda x: tag in x.tag_ids) instead of search